src/WorkflowManager/Database/Repositories/TasksRepository.cs

```diff
@@ -76,34 +76,34 @@ private static async Task EnsureIndex(IMongoCollection<WorkflowInstance> workflo
             }
         }
 
-        public async Task<IList<TaskExecution>> GetAllAsync(int? skip, int? limit)
+        public async Task<(IList<TaskExecution>, long)> GetAllAsync(int? skip, int? limit)
         {
             try
             {
-                var builder = Builders<WorkflowInstance>.Filter;
-
-                var filter = builder.Eq("Tasks.Status", TaskExecutionStatus.Accepted);
-                filter &= builder.Ne("Tasks.Status", TaskExecutionStatus.Dispatched);
+                var filter = new FilterDefinitionBuilder<WorkflowInstance>().Where(x =>
+                x.Status != Status.Succeeded && x.Status != Status.Failed);
```

Comment:
are you missing canceled check here?

Reply:
This is for the workflowinstances we don't have cancelled for these. I'm checking cancelled for the tasks below however.